internal/pkg/agent/application/paths/paths.go

```diff
@@ -17,6 +17,9 @@ const (
 	// ControlSocketPath is the control socket path used when installed.
 	ControlSocketPath = "unix:///run/elastic-agent.sock"
 
+	// ControlSocketNonRootPath is the control socket path used when installed as non-root.
+	ControlSocketNonRootPath = "unix:///run/elastic-agent/elastic-agent.sock"
```

Comment:
Why does this need a containing directory when not running as root? Because of the permissions on /run?

Reply:
Correct. The `/run/elastic-agent` directory must be created during install when executing as root, and then the `elastic-agent` user can write to that directory.

Reply:
Added comment on why it must be in a directory.